utils/get-us-data.R

```diff
@@ -1,78 +1,77 @@
-# anomaly_threshold = minimum death count above which to adjust anomalies (e.g 100)
-# check_adjustmente = boolean, if TRUE prints which states have had data adjusted
+library(magrittr)
+library(dplyr)
+library(tidyr)
+library(readr)
+library(lubridate)
   
-# Deaths data -------------------------------------------------------------
-get_us_deaths <- function(data = "daily"){  
+get_us_data <- function(data = c("deaths", "cases"),
+                        include_national = FALSE,
+                        incident = TRUE, 
+                        incident_weekly = FALSE) {  
+  
+  if (data == "cases") {
+    url <- "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_US.csv"
+  } else if (data == "deaths") {
+    url <- "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_US.csv"
+  }
 
-   # Get & reshape data
-   cumulative <- suppressMessages(readr::read_csv("https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_US.csv")) %>% 
-      dplyr::select(Province_State, dplyr::matches("^\\d")) %>%
-      tidyr::pivot_longer(cols = -Province_State, names_to = "date", values_to = "deaths") %>%
-      dplyr::mutate(date = lubridate::mdy(date)) %>%
-      dplyr::group_by(Province_State, date) %>%
-      dplyr::summarise(deaths = sum(deaths), .groups = "drop_last") %>%
-      dplyr::rename(state = Province_State) %>%
-      dplyr::mutate(epiweek = lubridate::epiweek(date)) %>%
-      dplyr::arrange(date) %>%
-      dplyr::filter(!state %in% c("Diamond Princess", "Grand Princess"))
+   # Get data & reshape from wide to long by date
+   cumulative <- suppressMessages(read_csv(url)) %>% 
+      select(Province_State, matches("^\\d")) %>%
+      pivot_longer(cols = -Province_State, names_to = "date", values_to = "value") %>%
+      mutate(date = mdy(date)) %>%
+      group_by(Province_State, date) %>%
+      summarise(value = sum(value), .groups = "drop_last") %>%
+      rename(state = Province_State) %>%
+      arrange(date) %>%
+      filter(!state %in% c("Diamond Princess", "Grand Princess"))
 
-   # Convert to daily data
-   daily <- cumulative %>%
-     dplyr::group_by(state) %>% 
-     dplyr::mutate(deaths = c(0, diff(deaths)),
-                   deaths = replace(deaths, deaths < 0 , 0)) %>%
-     dplyr::ungroup()
-     
-# Return data
-     if(data == "daily"){
-     return(daily)
+   # Arg: include US national sum of states as a "state"
+   if (include_national) {
+     national <- cumulative %>%
+       group_by(date) %>%
+       summarise(value = sum(value), .groups = "drop") %>%
+       mutate(state = "US")
+     cumulative <- bind_rows(cumulative, national)
    }
    
-   if(data == "cumulative"){
+   # Add epi dates
+   cumulative <- cumulative %>%
+     mutate(epiweek = epiweek(date),
```

Comment:
again epiweek and epiyear - do you think we should consider adding these to covidregional data?